gpflow/conditionals/util.py

```diff
@@ -30,14 +31,14 @@ def base_conditional(Kmn: tf.Tensor,
     :param Knn: [..., N, N]  or  N
     :param f: [M, R]
     :param full_cov: bool
-    :param q_sqrt: None or [R, M, M] (lower triangular)
+    :param q_sqrt: None or [R, M, M] (lower triangular) or [M, R] (diagonal)
```

Comment:
I'd suggest adding a type hint to `q_sqrt`. I guess it's `Optional[tf.Tensor]`. Then you can say
```
     :param q_sqrt: If not `None`, must have shape [R, M, M] (lower triangular) or [M, R] (diagonal) 
```
which I think reads a little more clearly. For example, isn't `None` a shape in some contexts?